### PR TITLE
Pin down LXD and use dqlite LTS for microcluster

### PR DESCRIPTION
**Pull request by roosterfish**
This PR makes sure we always only pull the LXD dependency from `@stable-5.21` and use microcluster `v2.0.5` which contains the dqlite LTS.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,12 +5,12 @@ go 1.22.7
...
 	github.com/AlecAivazis/survey/v2 v2.3.7
 	github.com/Netflix/go-expect v0.0.0-20220104043353-73e0943537d2
-	github.com/canonical/lxd v0.0.0-20241022112222-538639622c38
+	github.com/canonical/lxd v0.0.0-20241106123323-f0be5c8f36a6
 	github.com/canonical/microceph/microceph v0.0.0-20241021172252-2dd856d64c62
-	github.com/canonical/microcluster/v2 v2.0.4
-	github.com/canonical/microovn/microovn v0.0.0-20241018151920-4b82d2983810
-	github.com/charmbracelet/lipgloss v0.13.0
-	github.com/creack/pty v1.1.23
+	github.com/canonical/microcluster/v2 v2.0.5-0.20241106132149-fddf6637346f
```
Comment (roosterfish) [inline go.mod:10]: TODO: Change to the new tag once released.

**Comment by roosterfish** [comment]
@masnax looks one more dependency went into stable right before our release, see https://github.com/charmbracelet/lipgloss/releases/tag/v1.0.0

**Review by tomponline — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -5,12 +5,12 @@ go 1.22.7
...
 	github.com/Netflix/go-expect v0.0.0-20220104043353-73e0943537d2
-	github.com/canonical/lxd v0.0.0-20241022112222-538639622c38
+	github.com/canonical/lxd v0.0.0-20241106165613-4aab50ec18c3
 	github.com/canonical/microceph/microceph v0.0.0-20241021172252-2dd856d64c62
-	github.com/canonical/microcluster/v2 v2.0.4
-	github.com/canonical/microovn/microovn v0.0.0-20241018151920-4b82d2983810
-	github.com/charmbracelet/lipgloss v0.13.0
-	github.com/creack/pty v1.1.23
+	github.com/canonical/microcluster/v2 v2.0.5
+	github.com/canonical/microovn/microovn v0.0.0-20241101125123-0d5d663f6575
```
Comment (tomponline) [inline go.mod:11]: we should update the makefile update-gomod to target microovn and microceph LTS branches

**Comment by masnax** [comment]
> @masnax looks one more dependency went into stable right before our release, see https://github.com/charmbracelet/lipgloss/releases/tag/v1.0.0

This is from `microcloud status`.

**Comment by masnax** [comment]
Has this been tested yet to ensure dqlite leader detection is working okay? We may need to tweak the default heartbeat interval and max concurrent leader connections.

**Pull request merged**